e2e/cypress/integration/channel/channel_settings_spec.js

```diff
@@ -52,4 +54,32 @@ describe('Channel Settings', () => {
             cy.get('@channel').find('span.btn-close').should('exist');
         });
     });
+
+    it('MM-T882 Channel settings / Channel URL validation works properly', () => {
+        // # Login as admin and create new channel
+        cy.apiAdminLogin();
+        cy.apiCreateChannel(testTeam.id, 'coolest-channel', 'coolest channel').then(({channel}) => {
+            // # Visit the channel
+            cy.visit(`/${testTeam.name}/channels/${channel.name}`);
+
+            // # Go to channel dropdown > Rename channel
+            cy.get('#channelHeaderDropdownIcon').click();
+            cy.findByText('Rename Channel').click();
+
+            // # Try to enter existing URL and save
+            cy.get('#channel_name').clear().type('town-square');
+            cy.get('#save-button').click();
+
+            // # Error is displayed and URL is unchanged
+            // expect('.control-label').to.be.visible.and.to.eq('');
+            cy.url().should('include', `/${testTeam.name}/channels/${channel.name}`);
+
+            // # Enter a new URL and save
+            cy.get('#channel_name').clear().type('another-town-square');
+            cy.get('#save-button').click();
+
+            // URL is updated and no errors are displayed
+            cy.url().should('include', `/${testTeam.name}/channels/another-town-square`);
```

Comment:
Here, please add check that the error does not exist.
```suggestion
            cy.get('.has-error').should('not.exist');
            cy.url().should('include', `/${testTeam.name}/channels/another-town-square`);
```